diff --git a/BUILD.gn b/BUILD.gn
index 7c48268a24..2f5c8ff1d7 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -548,7 +548,9 @@ if (!build_with_chromium) {
     deps = [
       "api:create_modular_peer_connection_factory",
       "api:create_peerconnection_factory",
-      "api:enable_media",
+      "api:enable_media_with_defaults",
+      "modules/audio_device:test_audio_device_module",
       "api:rtc_error",
       "api:transport_api",
+      "api/video:adapted_video_track_source",
       "api/audio_codecs:opus_audio_decoder_factory",
diff --git a/call/rtp_config.cc b/call/rtp_config.cc
index 560ec099d6..207c5e392a 100644
--- a/call/rtp_config.cc
+++ b/call/rtp_config.cc
@@ -251,7 +251,8 @@ RtpStreamConfig RtpConfig::GetStreamConfig(size_t index) const {
   stream_config.raw_payload = raw_payload;
   if (!rtx.ssrcs.empty()) {
     RTC_DCHECK_EQ(ssrcs.size(), rtx.ssrcs.size());
-    auto& stream_config_rtx = stream_config.rtx.emplace();
+    auto& stream_config_rtx =
+        stream_config.rtx.emplace(RtpStreamConfig::Rtx{});
     stream_config_rtx.ssrc = rtx.ssrcs[index];
     stream_config_rtx.payload_type = rtx.payload_type;
   }
diff --git a/rtc_base/ssl_stream_adapter.h b/rtc_base/ssl_stream_adapter.h
index 0568bd0310..a6df194919 100644
--- a/rtc_base/ssl_stream_adapter.h
+++ b/rtc_base/ssl_stream_adapter.h
@@ -129,7 +129,7 @@ class SSLStreamAdapter : public StreamInterface {
   static std::unique_ptr<SSLStreamAdapter> Create(
       std::unique_ptr<StreamInterface> stream,
       absl::AnyInvocable<void(SSLHandshakeError)> handshake_error,
-      nullptr_t /*field_trials*/) {
+      std::nullptr_t /*field_trials*/) {
     return Create(std::move(stream), std::move(handshake_error));
   }
 
