diff --git a/call/payload_type_picker.cc b/call/payload_type_picker.cc
index 46cb5dc259..7e5693a466 100644
--- a/call/payload_type_picker.cc
+++ b/call/payload_type_picker.cc
@@ -368,7 +368,8 @@ void PayloadTypeRecorder::Rollback() {
 RTCError RtpHeaderExtensionRecorder::AddMapping(RtpHeaderExtensionId id,
                                                 absl::string_view uri,
                                                 bool encrypt) {
-  auto it = uri_to_id_.find(std::pair{uri, encrypt});
+  auto it =
+      uri_to_id_.find(std::pair<std::string, bool>(std::string(uri), encrypt));
   if (it != uri_to_id_.end()) {
     if (it->second != id) {
       RTC_HISTOGRAM_BOOLEAN(
@@ -391,7 +392,8 @@ RTCError RtpHeaderExtensionRecorder::AddMapping(RtpHeaderExtensionId id,
 RTCErrorOr<RtpHeaderExtensionId> RtpHeaderExtensionRecorder::LookupId(
     absl::string_view uri,
     bool encrypt) const {
-  auto it = uri_to_id_.find(std::pair{uri, encrypt});
+  auto it =
+      uri_to_id_.find(std::pair<std::string, bool>(std::string(uri), encrypt));
   if (it == uri_to_id_.end()) {
     return RTCError(RTCErrorType::INVALID_PARAMETER,
                     "No ID found for extension");
diff --git a/call/rtp_config.cc b/call/rtp_config.cc
index 560ec099d6..207c5e392a 100644
--- a/call/rtp_config.cc
+++ b/call/rtp_config.cc
@@ -251,7 +251,8 @@ RtpStreamConfig RtpConfig::GetStreamConfig(size_t index) const {
   stream_config.raw_payload = raw_payload;
   if (!rtx.ssrcs.empty()) {
     RTC_DCHECK_EQ(ssrcs.size(), rtx.ssrcs.size());
-    auto& stream_config_rtx = stream_config.rtx.emplace();
+    auto& stream_config_rtx =
+        stream_config.rtx.emplace(RtpStreamConfig::Rtx{});
     stream_config_rtx.ssrc = rtx.ssrcs[index];
     stream_config_rtx.payload_type = rtx.payload_type;
   }
diff --git a/rtc_base/ssl_stream_adapter.h b/rtc_base/ssl_stream_adapter.h
index 0568bd0310..a6df194919 100644
--- a/rtc_base/ssl_stream_adapter.h
+++ b/rtc_base/ssl_stream_adapter.h
@@ -129,7 +129,7 @@ class SSLStreamAdapter : public StreamInterface {
   static std::unique_ptr<SSLStreamAdapter> Create(
       std::unique_ptr<StreamInterface> stream,
       absl::AnyInvocable<void(SSLHandshakeError)> handshake_error,
-      nullptr_t /*field_trials*/) {
+      std::nullptr_t /*field_trials*/) {
     return Create(std::move(stream), std::move(handshake_error));
   }
 
